### DISCUSSION TITLE
Advice on concurrency value

### DISCUSSION THREAD

**Discussion by Rory-Powell** [OptimalBits/bull#2412] [General]
Is there any advice on what to set the concurrency to when using `queue.process()`? i.e.
- What is the recommended value to set in relation to the system resources available?
- What are the downsides to setting this value to a high number? say 50-100

**Comment by manast** [comment]
The best is actually to make some tests to find out a sweet spot. Normally jobs that are just IO intensive can be parallelized a lot in NodeJS, so having a high concurrency like 300 is not unusual, but if the jobs are more CPU intensive you would probably lower this value a lot and maybe even use sandboxed processors where the jobs will run in separate OS processes.

**Reply by Rory-Powell** [reply]
Thanks for the feedback @manast !
